### PR TITLE
Update utils.js

### PR DESCRIPTION
**Pull request by Th0mYT**
- Add function 'getFileNameFromFilePath' to extract the file name from the loadChildren path.
  Used to set the name of the chunks in the webpack output config: chunkFilename: '[name].chunk.js'.
- Update testing

### CODE REVIEW

**Comment by brandonroberts** [comment]
Thanks for the PR! I think this would be more flexible as a query string option in the `loadChildren` string. Its too specific to look for a `.module` in the filename. What do you think?

**Comment by Th0mYT** [comment]
Thanks for the reply. I think that I had not considered this option. you give an example of how you intend to implement it?

**Comment by brandonroberts** [comment]
Sure. Similar to the `sync=true` option.

```js
loadChildren: './path/to/lazy.module#LazyModule?chunkName=MyChunk'
```

replaced with

```js
loadChildren: () => new Promise(function (resolve) {
    (require as any).ensure([], function (require: any) {
      resolve(require('./path/to/lazy.module')['LazyModule']);
    });
  }, 'MyChunk')
```

**Comment by Th0mYT** [comment]
Not a bad idea, but this follows the angular approach?

The official documentation says:
"A lazy loaded module location is a string, not a type. In this app, the string identifies both the module file and the module class, the latter separated from the former by a #."

From what, add a query parameter might be the proper way?

Thank you.

**Comment by brandonroberts** [comment]
It still follows the angular approach and lazy load string convention. Chunk filenames are specific to webpack.

**Comment by Th0mYT** [comment]
Perfect. I will edit the source as soon as possible. Thanks you.

**Comment by Th0mYT** [comment]
I have update the source. Now should be find. Thanks you.

File: spec/utils.spec.js [hunk 1]
```diff
@@ -35,11 +35,10 @@ describe('Utils', function() {
         'loadChildren: () => new Promise(function (resolve) {',
         '  (require as any).ensure([], function (require: any) {',
         '    resolve(' + getRequireString(path, name) + ');',
-        '  });',
+        '  }, \'name\');',
         '})'
       ];
-
-      getRequireLoader('path', 'name', true).should.eql(result.join(''));
+      getRequireLoader('path', 'name', 'name', true).should.eql(result.join(''));
```
Comment (brandonroberts) [inline spec/utils.spec.js]: Add this as a separate test so we have one with/without a chunk name.

**Comment by brandonroberts** [comment]
Resolve merge conflicts and add the separate test and this one should be good to go.

**Comment by brandonroberts** [comment]
Thanks!

**Pull request merged**